### PR TITLE
Add option for sessionModelName

### PR DESCRIPTION
**Pull request by jeremygottfried**
eg usage:

```js
session({
  secret: process.env.COOKIE_SECRET || 'secret',
  store: new PrismaSessionStore(
     prisma,
     {
       sessionModelName: 'oAuthSession'
     }
   ),
  cookie: { maxAge: 7 * 24 * 60 * 60 * 1000 } // 1 week
})
```

### CODE REVIEW

**Review by kleydon — CHANGES_REQUESTED** [review]
Hi @jeremygottfried, appreciate your PR.

I just tried to run the test suite using `yarn test` (can also use `npm run test`), and its not yet working for me...
Would you consider updating this, so that the tests pass? Guessing the issue is on line 29 of src/lib/prisma-session-store.spec.ts.

Best,

-K

File: src/@types/prisma.ts [hunk 1]
```diff
@@ -43,15 +43,16 @@ interface IDeleteArgs {
   where: { sid: string };
 }
 
-export interface IPrisma {
-  session: {
+export type IPrisma = {
+  [key: string]: {
```
[the remark this answers is not in the document]
Comment (kleydon) [reply src/@types/prisma.ts:47]: @wSedlacek - Care to weigh in on this?

[the remark this answers is not in the document]
Comment (wSedlacek) [reply src/@types/prisma.ts:47]: We are losing some type safety by making this an index.

I would prefer if the key was using a generic tied to the new option.

[the remark this answers is not in the document]
Comment (wSedlacek) [reply src/@types/prisma.ts:47]: Looks like the index is the source of the test failing too.

**Comment by jeremygottfried** [comment]
> Hi @jeremygottfried, appreciate your PR.
> 
> I just tried to run the test suite using `yarn test` (can also use `npm run test`), and its not yet working for me...
> Would you consider updating this, so that the tests pass? Guessing the issue is on line 29 of src/lib/prisma-session-store.spec.ts.
> 
> Best,
> 
> -K

Yeah I'll try to get to it this weekend.

**Comment by wSedlacek** [comment]
@jeremygottfried Would you mind if I updated the types for you?
The getting the generic to infer correctly can be a bit tricky and I got a few hours to spare this week.

**Comment by jeremygottfried** [comment]
@wSedlacek yeah that would be great thanks!

**Comment by wSedlacek** [comment]
Here is a PR into your PR - https://github.com/jeremygottfried/prisma-session-store/pull/1

**Review by wSedlacek — APPROVED** [review]
This looks good to me.
Test are passing.

**Review by kleydon — APPROVED** [review]

File: src/@types/options.ts [hunk 2]
```diff
@@ -91,6 +91,12 @@ export interface IOptions {
    */
   ttl?: number | TTLFactory;
 
+  /**
+   * "Session Table Name"
+   * defines session table name. Defaults to sessions
+   */
+  sessionModelName?: Exclude<M, `$${string}`>;
```
Comment (kleydon) [inline src/@types/options.ts:98]: @jeremygottfried, @wSedlacek , can you help me understand how generics are working here? Specifically, I'm not following the `$${string}`. (Thanks!)

**Review by kleydon — CHANGES_REQUESTED** [review]
Had one idea to retain some test coverage, and another question. Overall, looks solid.

File: src/@types/options.ts (hunk 2 above)
Comment (wSedlacek) [reply src/@types/options.ts:98]: The `$${string}` here is looking for any string that starts with a `$`, ie `$use` or `$on` which are used in Prisma to denote internal methods.

The reason it is used here is to make sure the `IPrisma` interface does not infer that `$use` (and the other internal methods) should have a `.create()` method.

File: tests/integration.spec.ts [hunk 3]
```diff
@@ -32,6 +32,7 @@ describe('integration testing', () => {
         saveUninitialized: false,
         store: new PrismaSessionStore(prisma, {
           logger: false,
+          sessionModelName: 'otherSession',
```
[the remark this answers is not in the document]
Comment (wSedlacek) [reply tests/integration.spec.ts:35]: I updated these test.
The testing without a sessionModelName is covered by the unit testing.
I am not against adding it here, but it wouldn't affect test coverage.

File: src/@types/options.ts (hunk 2 above)
Comment (kleydon) [reply src/@types/options.ts:98]: Ah - got it. Thanks.

File: tests/integration.spec.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (kleydon) [reply tests/integration.spec.ts:35]: 👍 Just wanted to ensure there was coverage somewhere.

**Pull request merged**